addons/web_editor/models/ir_qweb_fields.py

```diff
@@ -71,6 +71,10 @@ def _compile_node(self, el, compile_context, indent):
     def _compile_directive_snippet(self, el, compile_context, indent):
         key = el.attrib.pop('t-snippet')
         el.set('t-call', key)
+        snippet_lang = self._context.get('snippet_lang')
+        if snippet_lang:
+            el.set('t-lang', f"'{snippet_lang}'")
```

Comment:
You could have probably used `"%r" % snippet_lang` that would have handled the quote itself